### PR TITLE
Add tests to model/deployment_chain

### PR DESCRIPTION
**Pull request by kentakozuka**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

- **How are users affected by this change**:
- **Is this breaking change**:
- **How to migrate (if breaking change)**:

### CODE REVIEW

File: pkg/model/deployment_chain_test.go [hunk 1]
```diff
@@ -110,3 +110,239 @@ func TestDeploymentChainDesireStatus(t *testing.T) {
 		})
 	}
 }
+func TestListAllInChainApplicationDeploymentsMap(t *testing.T) {
+	dc := &DeploymentChain{
```
Comment (ffjlabo) [inline pkg/model/deployment_chain_test.go:114]: Let's add `t.Parallel` !

File: pkg/model/deployment_chain_test.go [hunk 2]
```diff
@@ -110,3 +110,239 @@ func TestDeploymentChainDesireStatus(t *testing.T) {
...
+		"app2": {DeploymentId: "dep2"},
+		"app3": {DeploymentId: "dep3"},
+	}
+
+	got := dc.ListAllInChainApplicationDeploymentsMap()
+
+	assert.Equal(t, want, got)
+}
+func TestListAllInChainApplications(t *testing.T) {
+	dc := &DeploymentChain{
```
Comment (ffjlabo) [inline pkg/model/deployment_chain_test.go:154]: also add t.Parallel 👍

File: pkg/model/deployment_chain_test.go [hunk 3]
```diff
@@ -110,3 +110,239 @@ func TestDeploymentChainDesireStatus(t *testing.T) {
...
+	}
+
+	got := dc.ListAllInChainApplications()
+
+	assert.Equal(t, want, got)
+}
+
+func TestIsCompleted(t *testing.T) {
+	tests := []struct {
+		name   string
```
Comment (ffjlabo) [inline pkg/model/deployment_chain_test.go:197]: also add t.Parallel 👍

File: pkg/model/deployment_chain_test.go [hunk 4]
```diff
@@ -110,3 +110,239 @@ func TestDeploymentChainDesireStatus(t *testing.T) {
...
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			b := &ChainBlock{Status: tt.status}
+			got := b.IsCompleted()
+			assert.Equal(t, tt.want, got)
+		})
+	}
+}
+func TestDesiredStatus(t *testing.T) {
+	tests := []struct {
```
Comment (ffjlabo) [inline pkg/model/deployment_chain_test.go:237]: also add t.Parallel 👍

File: pkg/model/deployment_chain_test.go [hunk 5]
```diff
@@ -110,3 +110,239 @@ func TestDeploymentChainDesireStatus(t *testing.T) {
...
+		t.Run(tt.name, func(t *testing.T) {
+			gotDesired := tt.block.DesiredStatus()
+			assert.Equal(t, tt.wantDesired, gotDesired)
+		})
+	}
+}
+
+func TestGetNodeByDeploymentID(t *testing.T) {
+	block := &ChainBlock{
+		Nodes: []*ChainNode{
```
Comment (ffjlabo) [inline pkg/model/deployment_chain_test.go:302]: also add t.Parallel 👍

File: pkg/model/deployment_chain_test.go [hunk 6]
```diff
@@ -110,3 +110,239 @@ func TestDeploymentChainDesireStatus(t *testing.T) {
...
+		{
+			name:   "returns false for DEPLOYMENT_BLOCK_RUNNING",
+			status: ChainBlockStatus_DEPLOYMENT_BLOCK_RUNNING,
+			want:   false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			b := &ChainBlock{Status: tt.status}
```
Comment (ffjlabo) [inline pkg/model/deployment_chain_test.go:230]: it should be better to add t.Parallel and `tt := tt` 👍

File: pkg/model/deployment_chain_test.go [hunk 7]
```diff
@@ -110,3 +110,239 @@ func TestDeploymentChainDesireStatus(t *testing.T) {
...
+			name:         "returns error for nil deployment ref",
+			deploymentID: "",
+			wantNode:     nil,
+			wantErr:      true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gotNode, err := block.GetNodeByDeploymentID(tt.deploymentID)
```
Comment (ffjlabo) [inline pkg/model/deployment_chain_test.go:343]: Also here
> it should be better to add t.Parallel and tt := tt 👍

File: pkg/model/deployment_chain_test.go [hunk 8]
```diff
@@ -110,3 +110,239 @@ func TestDeploymentChainDesireStatus(t *testing.T) {
...
+			block: &ChainBlock{
+				Status: ChainBlockStatus_DEPLOYMENT_BLOCK_PENDING,
+			},
+			wantDesired: ChainBlockStatus_DEPLOYMENT_BLOCK_SUCCESS,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gotDesired := tt.block.DesiredStatus()
```
Comment (ffjlabo) [inline pkg/model/deployment_chain_test.go:294]: Also here
> it should be better to add t.Parallel and tt := tt 👍

**Review by ffjlabo — COMMENTED** [review]
Thank you for adding unit tests! It's so helpful!
I added some comments!

**Comment by kentakozuka** [comment]
@ffjlabo Fixed 👍

**Review by khanhtc1202 — APPROVED** [review]
🚀

**Pull request merged**